src/Microsoft.TemplateEngine.Utils/TemplateEngineEventSource.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.TemplateEngine.Utils
+{
+    /// <summary>
+    /// This captures information of how various key methods of TemplateEngine ran.
+    /// </summary>
+    [EventSource(Name = "Microsoft-TemplateEngine")]
+    public sealed class TemplateEngineEventSource : EventSource
```

Comment:
Couple of thoughts:
- is there any way to make logging via parameter instead of defining method for each event? Now you would need to add new method each time we want to log smth new.
- would it be better to split it and make internal to caller assemblies? (internal class for CLI and internal class for Edge)